cunumeric/thunk.py

```diff
@@ -179,6 +179,14 @@ def transpose(self, rhs, axes, stacklevel):
         """
         raise NotImplementedError("Implement in derived classes")
 
+    def choose(self, choices, out, mode, stacklevel):
```

Comment:
The signature of every method on `NumpyThunk` (defined in thunk.py) should be mirrored on its subclasses in deferred.py, eager.py and lazy.py. Currently the signature on deferred.py is different. You probably want to change the implementations on all the other files to match the one on deferred.py.

Note that the thunk method is the lower-level implementation, and is only supposed to be called from the higher-level method in array.py or module.py. Therefore, it can have narrower functionality than the full operation.

Reply:
Ok, understood. I will make sure the signature in the same in deferred.py, eager.py and lazy.py